crates/ra_ide/src/inlay_hints.rs

```diff
@@ -59,6 +65,46 @@ pub(crate) fn inlay_hints(
     res
 }
 
+fn get_chaining_hints(
+    acc: &mut Vec<InlayHint>,
+    sema: &Semantics<RootDatabase>,
+    options: &InlayHintsOptions,
+    expr: ast::Expr,
+) -> Option<()> {
+    if !options.chaining_hints {
+        return None;
+    }
+
+    let ty = sema.type_of_expr(&expr)?;
+    let label = ty.display_truncated(sema.db, options.max_length).to_string();
```

Comment:
I'd suggest moving this variable declaration closer to its usage. Probably right under the `if` at the end of the method, so we don't compute this string if we cannot create the chaining hint. What do you think?

Reply:
Good point. Done.